src/iotEdgeExplorer.ts

```diff
@@ -102,6 +114,40 @@ export class IoTEdgeExplorer extends BaseExplorer {
         }
     }
 
+    private async isValidDeploymentJson(jsonStr: string): Promise<boolean> {
+        try {
+            const schema = (await axios.get(Constants.DeploymentJsonSchemaUrl)).data;
+            const ajv = new Ajv({ allErrors: true });
+
+            const json = this.tryParseJson(jsonStr);
+            if (json === null) {
+                TelemetryClient.sendEvent(Constants.IoTHubAIValidateJsonSchemaEvent, { error: "The deployment json file is not a valid json file" });
+                vscode.window.showErrorMessage("The deployment json file is not a valid json file");
+                return false;

```

Comment:
how about just throw error here. and the telemetry of error and error message could be sent in catch block.

Reply:
When deployment file is not a json string, then we will **not continue** to create deployment, so it return false.

However, when we cannot validate json schema ( for example: user's network is poor, or some internal exception throw by the validator), and we will **continue** to create deployment, so it return true.

I think this two types of  error cannot combined together.

Reply:
Yes, better to different types of error and in error handling we could handle different types of errors. The sendEvent telemetry in lots of if/else block not looks clean.